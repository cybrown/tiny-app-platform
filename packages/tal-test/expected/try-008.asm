main()
  entry:
    FunctionRef       name: oneLinerTry_0
    DeclareLocal      name: oneLinerTry, mutable: false, hasInitialValue: true
oneLinerTry_0()
  entry:
    Try               catchLabel: try_catch_1, endTryLabel: try_end_2
    Literal           0
    MakeArray
    Literal           "arg"
    Local             arg1
    Literal           1
    MakeRecord
    Local             some_function_that_throws
    Call
    TryPop
    Jump              try_end_2
  try_catch_1:
    ScopeEnter
    PushLatestError
    DeclareLocal      mutable: false, name: err, hasInitialValue: true
    Pop               inBlock: false
    Local             error
    Literal           1
    MakeArray
    Literal           0
    MakeRecord
    Local             log
    Call
    ScopeLeave        inBlock: false
    Jump              try_end_2
  try_end_2:
    MakeArrayForBlock count: 1
