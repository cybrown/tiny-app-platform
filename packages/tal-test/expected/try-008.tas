fun oneLinerTry() { some_function_that_throws(arg: arg1) catch log(error) }
