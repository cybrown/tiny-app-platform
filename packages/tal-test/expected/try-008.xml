<program>
  <declarelocal>
    <function>
      <block>
        <try>
          <call>
            <local />
            <local />
          </call>
          <call>
            <local />
            <local />
          </call>
        </try>
      </block>
    </function>
  </declarelocal>
</program>
